xbmc/interfaces/json-rpc/PlayerOperations.cpp

```diff
@@ -361,6 +365,72 @@ JSONRPC_STATUS CPlayerOperations::Stop(const std::string &method, ITransportLaye
   }
 }
 
+JSONRPC_STATUS CPlayerOperations::GetAudioDelay(const std::string& method,
+                                                ITransportLayer* transport,
+                                                IClient* client,
+                                                const CVariant& parameterObject,
+                                                CVariant& result)
+{
+  auto& components = CServiceBroker::GetAppComponents();
+  const auto appPlayer = components.GetComponent<CApplicationPlayer>();
+  result["offset"] = appPlayer->GetVideoSettings().m_AudioDelay;
+  return OK;
+}
+
+JSONRPC_STATUS CPlayerOperations::SetAudioDelay(const std::string& method,
+                                                ITransportLayer* transport,
+                                                IClient* client,
+                                                const CVariant& parameterObject,
+                                                CVariant& result)
+{
+  switch (GetPlayer(parameterObject["playerid"]))
+  {
+    case Video:
+    {
+      auto& components = CServiceBroker::GetAppComponents();
+      const auto appPlayer = components.GetComponent<CApplicationPlayer>();
+      float videoAudioDelayRange =
+          CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_videoAudioDelayRange;
+
+      if (parameterObject["offset"].isDouble())
+      {
+        float offset = static_cast<float>(parameterObject["offset"].asDouble());
+        offset = MathUtils::RoundF(offset, 0.025f);
+        if (offset <= videoAudioDelayRange || offset >= -videoAudioDelayRange)
+          appPlayer->SetAVDelay(offset);
+      }
+      else if (parameterObject["offset"].isString())
+      {
+        CVideoSettings vs = appPlayer->GetVideoSettings();
+        if (parameterObject["offset"].asString().compare("increment") == 0)
+        {
+          vs.m_AudioDelay += 0.025f;
```

Comment:
magic number for the step

Reply:
also `ACTION_AUDIO_DELAY`, `ACTION_AUDIO_DELAY_MIN`, `ACTION_AUDIO_DELAY_PLUS` seem to use the same step. Would be great to just have this as a constant somewhere to avoid hypothetical inconsistencies in the future.

Reply:
added to video/dialogs/GUIDialogAudioSettings.h
seem there is no good place
if you have other suggests let me know

Reply:
Sorry, as it is shared code it can't be in some gui component. otherwise you include gui stuff everywhere.
This is not a mandatory change, just a recommendation, you're free to keep the magic numbers as before

Reply:
if i place it to https://github.com/xbmc/xbmc/blob/master/xbmc/video/VideoUtils.h?

Reply:
Refers to audio, not video. Maybe settings/MediaSettings.h ?

Reply:
yeah better than nothing